bundles/org.eclipse.e4.ui.css.swt.theme/src/org/eclipse/e4/ui/css/swt/internal/theme/ThemeEngine.java

```diff
@@ -607,7 +618,7 @@ public void restore(String alternateTheme) {
 				 * this case want to fall back to respect whether that previous choice was dark
 				 * or not. https://github.com/eclipse-platform/eclipse.platform.ui/issues/2776
 				 */
-				overrideWithDarkTheme = prefThemeId.contains("dark");
+				overrideWithDarkTheme = getPreferenceThemeIsDark();
```

Comment:
here we lose the fallback in case the explicit new isDark preference is not (yet) set in an older installation.
I think we should let `getPreferenceThemeIsDark()` return a `Boolean` and in case of `null`, fall back to the heuristics.

Reply:
But how do we knot if the preference is not (yet) set?
JavaDoc of the get-methods say:

> Returns the value associated with the specified key in this node. Returns the specified default if there is no value associated with the key, or the backing store is inaccessible.

Reply:
Ah, then something like:

```java
	private boolean getPreferenceThemeIsDark(boolean fallback) {
		return getPreferences().getBoolean(THEME_IS_DARK_KEY, fallback);
	}
```

```java
    ...
    overrideWithDarkTheme = getPreferenceThemeIsDark(prefThemeId.contains("dark"));
    ....
```

Reply:
ah sure. That's a good idea.

Reply:
see 8fe3319